packages/hash/frontend/src/components/hooks/blockProtocolFunctions/useBlockProtocolDeleteLink.ts

```diff
@@ -0,0 +1,71 @@
+import { useMutation } from "@apollo/client";
+
+import { EmbedderGraphMessageCallbacks } from "@blockprotocol/graph";
+import { deleteLinkMutation } from "@hashintel/hash-shared/queries/link.queries";
+import { useCallback } from "react";
+import {
+  DeleteLinkMutation,
+  DeleteLinkMutationVariables,
+} from "../../../graphql/apiTypes.gen";
+import { parseLinkIdentifier } from "../../../lib/entities";
+
+export const useBlockProtocolDeleteLink = (): {
```

Comment:
These hooks all feel very template-y / copy & paste-y. Do they each individually need to be hooks, or can there be one `useGraphMethods` and each of these functions can be properties on an object returned and can be generated?

Reply:
Some of the repetition can be eliminated - I've created [a task](https://app.asana.com/0/1201095311341924/1202584610589613/f) to do so